scripts/scheduled_posts.coffee

```diff
@@ -15,21 +15,18 @@ switch process.env.NODE_ENV
 # Connect to database
 db = mongojs(process.env.MONGOHQ_URL, ['articles'])
 
-db.articles.update(
-  { scheduled_publish_at: { $lt: new Date() } }
-  { 
-    $set: {
-      published: true
-      published_at: Date()
-      scheduled_publish_at: null 
-    } 
-  }
-  { multi: true }
-  (err, msg) ->
-    return exit err if err
-    debug 'Scheduled publication finished', msg
+db.articles.find(
+  scheduled_publish_at: { $lt: new Date() }
+).forEach (err, doc) ->
+  if !doc
+    debug 'Scheduled publication finished'
     process.exit()
-)
+  if err
+    exit err
+  doc.published = true
+  doc.published_at = moment(doc.scheduled_publish_at).toDate()
+  doc.scheduled_publish_at = null
+  db.articles.save doc
```

Comment:
Using `$set` with `Date()` sets the value of the document to the JS Date format. However in mongo, dates should be stored as the ISO string format. The problem was that some articles that were scheduled couldn't be queried by date because their `published_at` was in JS Date format. 

If the mongojs wrapper is used, it's fine to save using JS Date and the wrapper will convert and save it for us, but if you're using $set, it just sets it to the string without a transform, it seems. 

Another small change is to use `doc.scheduled_publish_at` as the new `published_at` date. Unfortunately with mongo, you can't perform an `update` while using the values of the current doc, you have to use a function transform instead which is why i'm doing `forEach`.